src/core/timer.c

```diff
@@ -391,12 +393,18 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
                 if (v->base == TIMER_CALENDAR) {
                         usec_t b, rebased;
 
-                        /* If we know the last time this was
-                         * triggered, schedule the job based relative
-                         * to that. If we don't, just start from
-                         * the activation time. */
-
-                        if (dual_timestamp_is_set(&t->last_trigger))
+                        /* If DeferReactivation is enabled, schedule
+                         * the job based on the last time the trigger unit
+                         * entered inactivity. Otherwise, if we know the
+                         * last time this was triggered, schedule the job
+                         * based relative to that. If we don't, just start
+                         * from the activation time or realtime. */
+
+                        if (t->defer_reactivation &&
+                                        dual_timestamp_is_set(&trigger->inactive_enter_timestamp) &&
+                                        dual_timestamp_is_set(&t->last_trigger))
+                                b = MAX(trigger->inactive_enter_timestamp.realtime, t->last_trigger.realtime);
```

Comment:
shouldn't there be another branch here that covers the case where last_trigger is unset, but we do have inactive_enter_timestamp? i.e.:

```c
if (t->defer_reactivation && dual_timestamp_is_set(&trigger->inactive_enter_timestamp)) {
        if (dual_timestamp_is_set(&t->last_trigger))
                b = MAX(trigger->inactive_enter_timestamp.realtime, t->last_trigger.realtime);
        else
                b = trigger->inactive_enter_timestamp;
} else if (dual_timestamp_is_set(&t->last_trigger)) 
        …
```